_posts/2024-04-02-shared-conda-tutorial.md

```diff
@@ -0,0 +1,103 @@
+
+On multi-user systems (e.g. shared Linux servers), multiple users can share the same conda installation and the environments thererin. This is a good idea and usually works well when only already existing environments are re-used, but can become tricky when users also want to create environments (using `conda create`) into the shared conda installation. On a standard Linux system, this often results in permission problems when (re-)downloading packages and overwriting files that were created by others.
+
+Here we describe a setup with a shared (read-only) conda installation, containing re-usable shared environments, and private per-user environments.
+
+## Tutorial
+
+### Creating the shared conda installation
+
+First, we create a shared conda installation folder that is read-only for all regular users. Shared conda environments therein, which are available to all users, must be created by a sysadmin (or by a user with write-privileges for the conda folder). 
+
+In this tutorial we'll use `/apps/conda` as the path, but please choose something appropriate for your system.
+
+We install conda by downloading and running the [Miniconda installer](https://repo.anaconda.com/miniconda/), but feel free to use the [full Anaconda installer](https://repo.anaconda.com/archive/) if that is your preference. We also set up folder permissions and an initial conda config, which includes the recommended [bioconda](https://bioconda.github.io/) and [conda-forge](https://conda-forge.org/) channels.
+
+```bash
+# Create shared conda folder
+CONDA_PATH="/apps/conda"
+sudo mkdir -pv "${CONDA_PATH}"
+sudo chmod 755 "${CONDA_PATH}"
+
+# Download Miniconda installer
+CONDA_URL="https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh"
+wget "${CONDA_URL}" --output-document=/tmp/conda_installer.sh
+
+# Install Miniconda
+# -b          run install in batch mode (without manual intervention),
+#             it is expected the license terms (if any) are agreed upon
+# -u         update an existing installation (needed to install in existing dir)
+# -p PREFIX  install prefix, defaults to $PREFIX, must not contain spaces.
+sudo bash /tmp/conda_installer.sh -b -u -p "${CONDA_PATH}"
+
+# Set default conda settings for all users
+# (these will show up in $CONDA_PATH/.condarc)
+sudo "${CONDA_PATH}"/bin/conda config --system --set channel_priority strict
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels defaults
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels bioconda
+sudo "${CONDA_PATH}"/bin/conda config --system --add channels conda-forge
+```
+
+Users can override the default conda settings in their own `$HOME/.condarc` configuration file, [(see more)](https://conda.io/projects/conda/en/latest/user-guide/configuration/admin-multi-user-install.html#user-configuration-file).
```

Comment:
The page linked here is not really helpful, I think. It describes a specific case that is different from the one in this post..

Reply:
Fair enough. Maybe it's distracting rather than helpful to see the official docs since it doesn't really cover exactly the same thing.

Reply:
So remove the whole sentence. There is nothing for a standard user to override in our scenaria..